Paintroid/src/main/java/org/catrobat/paintroid/dialog/PermanentDenialPermissionInfoDialog.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2015 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.paintroid.dialog;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.support.annotation.DrawableRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatDialogFragment;
+
+import org.catrobat.paintroid.R;
+
+public class PermanentDenialPermissionInfoDialog extends AppCompatDialogFragment {
+
+	private static final String PERMISSION_TYPE_KEY = "permissionTypeKey";
+	private static final String PERMISSIONS_KEY = "permissionsKey";
+	private static final String REQUEST_CODE_KEY = "requestCodeKey";
+
+	private PermissionType permissionType;
+
+	public static PermanentDenialPermissionInfoDialog newInstance(PermissionType permissionType, String[] permissions, int requestCode) {
+		PermanentDenialPermissionInfoDialog permissionInfoDialog = new PermanentDenialPermissionInfoDialog();
+
+		Bundle bundle = new Bundle();
+		bundle.putSerializable(PERMISSION_TYPE_KEY, permissionType);
+		bundle.putStringArray(PERMISSIONS_KEY, permissions);
+		bundle.putInt(REQUEST_CODE_KEY, requestCode);
+		permissionInfoDialog.setArguments(bundle);
+		return permissionInfoDialog;
+	}
+
+	@Override
+	public void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		Bundle arguments = getArguments();
+		permissionType = (PermissionType) arguments.getSerializable(PERMISSION_TYPE_KEY);
+	}
```

Comment:
There is only one type, which is `EXTERNAL_STORAGE`. What type do you expect in the argument?